src/test/java/uk/gov/di/ipv/service/UserIdentityServiceTest.java

```diff
@@ -7,18 +7,29 @@
 import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.di.ipv.persistence.DataStore;
 import uk.gov.di.ipv.persistence.item.UserIssuedCredentialsItem;
+import uk.org.webcompere.systemstubs.environment.EnvironmentVariables;
+import uk.org.webcompere.systemstubs.jupiter.SystemStub;
+import uk.org.webcompere.systemstubs.jupiter.SystemStubsExtension;
+import uk.org.webcompere.systemstubs.properties.SystemProperties;
 
 import java.time.LocalDateTime;
 import java.util.List;
 import java.util.Map;
 
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.Mockito.when;
+import static uk.gov.di.ipv.service.ConfigurationService.IS_LOCAL;
 
+@ExtendWith(SystemStubsExtension.class)
```

Comment:
:question: are we not able to refactor our implementations to make it so we don't _need_ to do this? I've had to deal with this in the past and it can make tests much slower, harder to parallelise and harder to maintain

Reply:
Yes, we should have a think about this.

Reply:
It doesn't feel ideal, but it does seem a convenient way to test code paths we can't inject into.
The tests are still pretty speedy....

Reply:
We say we can't inject - is that something we can solve by providing test-only constructors that allow us to inject mocks, then production code uses the current means?

Reply:
That's exactly what we're doing at the moment. These tests are to try to raise our code coverage by exercising those constructors used by the prod code. Been chatting with Kerr about some alternatives - possibly just ignoring them from the code coverage report.

Reply:
Ah gotcha, in which case I'd recommend just having a test which calls the constructor and does nothing else? Makes it clear its just for coverage / as you say just exclude it?

Reply:
i _think_ that's what i'm doing here? Thing is, the code the constructors call need certain env variables set - hence the library use.